src/main/scala/ai/privado/threatEngine/ThreatEngineExecutor.scala

```diff
@@ -123,33 +123,37 @@ class ThreatEngineExecutor(cpg: Cpg, dataflows: Map[String, Path], repoPath: Str
   }
 
   private def processDataflowViolations(threat: PolicyOrThreat) = {
-    val threatId = threat.id
-
-    // process android threats
-    val violationResponse = getAndroidManifestFile(repoPath) match {
-      // if we get a manifest file, consider android app
-      case Some(manifestFile) =>
-        logger.debug(s"Found AndroidManifest.xml: ${manifestFile}")
-        logger.info(s"Processing 'dataflow' threat: ${threatId}")
-        threatId match {
-          case "Threats.Leakage.isDataLeakingToLog" =>
-            DataLeakageToLogs.getViolations(threat, cpg, dataflows) match {
-              case Success(res) => Some(res)
-              case Failure(e)   => None
-            }
-          case "Threats.Sharing.isDataExposedToThirdPartiesViaNotification" =>
-            DataLeakageToNotifications.getViolations(threat, cpg, dataflows) match {
-              case Success(res) => Some(res)
-              case Failure(e)   => None
-            }
-
-          case _ =>
-            logger.debug(s"No implementation detected for threat: ${threatId}")
-            None
-        }
-      case _ =>
-        logger.debug("Did not find AndroidManifest.xml")
-        None
+    val threatId                                                             = threat.id
+    var violationResponse: Option[(Boolean, List[PolicyViolationFlowModel])] = None
+
+    // Data leakage to logs is threat on non Android repos too
+    if (threatId == "Threats.Leakage.isDataLeakingToLog") {
+      violationResponse = DataLeakageToLogs.getViolations(threat, cpg, dataflows) match {
+        case Success(res) => Some(res)
+        case Failure(e)   => None
+      }
+    } else {
```

Comment:
@abhstabs Can we handle this better in a generalized way?   
With more such threats, this will clutter with if-else-if-else statements.